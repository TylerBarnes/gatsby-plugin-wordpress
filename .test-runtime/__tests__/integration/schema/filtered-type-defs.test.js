import fetchGraphql from "gatsby-source-wordpress-experimental/utils/fetch-graphql"

describe(`[gatsby-source-wordpress-experimental] filtered type definitions`, () => {
  test(`Date field resolver is working`, async () => {
    const result = await fetchGraphql({
      url: `http://localhost:8000/___graphql`,
      query: /* GraphQL */ `
        {
          wpPage {
            year: date(formatString: "YYYY")
            month: date(formatString: "MMMM")
            dayOfMonth: date(formatString: "DD")
            dayOfWeekNumber: date(formatString: "E")
            dayOfWeekName: date(formatString: "dddd")
            date
          }
        }
      `,
    })

    expect(result).toMatchSnapshot()
  })
})
