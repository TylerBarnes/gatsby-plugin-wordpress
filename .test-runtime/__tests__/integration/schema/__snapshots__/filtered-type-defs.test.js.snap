// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[gatsby-source-wordpress-experimental] filtered type definitions Date field resolver is working 1`] = `
Object {
  "data": Object {
    "wpPage": Object {
      "date": "2020-03-19T01:41:08",
      "dayOfMonth": "19",
      "dayOfWeekName": "Thursday",
      "dayOfWeekNumber": "4",
      "month": "March",
      "year": "2020",
    },
  },
}
`;
